src/Stripe.net/Infrastructure/Public/StripeClient.cs

```diff
@@ -1,23 +1,51 @@
 namespace Stripe
 {
+    using System;
     using System.Net;
     using System.Net.Http;
     using System.Threading;
     using System.Threading.Tasks;
     using Newtonsoft.Json.Linq;
+    using Stripe.Infrastructure;
 
     /// <summary>
     /// A Stripe client, used to issue requests to Stripe's API and deserialize responses.
     /// </summary>
     public class StripeClient : IStripeClient
     {
         /// <summary>Initializes a new instance of the <see cref="StripeClient"/> class.</summary>
+        /// <param name="apiKey">The API key used by the client to make requests.</param>
+        /// <param name="clientId">The client ID used by the client in OAuth requests.</param>
         /// <param name="httpClient">
         /// The <see cref="IHttpClient"/> client to use. If <c>null</c>, an HTTP client will be
         /// created with default parameters.
         /// </param>
-        public StripeClient(IHttpClient httpClient = null)
+        /// <exception cref="ArgumentNullException">if <c>apiKey</c> is <c>null</c></exception>
+        /// <exception cref="ArgumentException">
+        /// if <c>apiKey</c> is empty or contains whitespace
+        /// </exception>
+        public StripeClient(
+            string apiKey,
+            string clientId = null,
+            IHttpClient httpClient = null)
         {
+            if (apiKey == null)
+            {
+                throw new ArgumentNullException(nameof(apiKey));
+            }
+
+            if (apiKey.Length == 0)
+            {
+                throw new ArgumentException("API key cannot be the empty string.", nameof(apiKey));
+            }
+
+            if (StringUtils.ContainsWhitespace(apiKey))
+            {
+                throw new ArgumentException("API key cannot contain whitespace.", nameof(apiKey));
+            }
+
```

Comment:
Should this logic be put into an API Key validator or something since I'm assuming you have the same in the config and potentially the same on the request options?

Reply:
We don't validate API keys in request options right now. We could, but the logic would be slightly different since `null` is allowed there.

The validation logic is essentially the same for the `StripeConfiguration. BuildDefaultStripeClient()` helper and the `StripeClient` constructor, but the exceptions that are thrown are different. The former throws `StripeException` with a long/helpful error message, the latter throws `ArgumentNullException` or `ArgumentException` with terse messages (which is more idiomatic, and I think fine since only "advanced" users creating their own client would see those).